cityofaustin/atd-moped
# Add council districts to project list

## Associated issues

This PR aims to close https://github.com/cityofaustin/atd-data-tech/issues/10892.

## TIL 

`unnest()` & `array_agg(DISTINCT)` are a powerful combination to roll up a distinct set of values.

## Testing

### Local testing instructions

* Spin up the local database, optionally pulling down a copy of production data for better testing.
* Apply migrations/metadata
* Query the `project_list_view` like so, and spot check council district data until you're confident this is working as intended. 
```sql
# edit, i forgot this when i first opened up this PR, sorry!

select project_id, project_description, self_alone_council_districts, self_and_children_council_districts
from project_list_view 
where project_id = ?
```
  * There are only a handful of projects which have more council district coverage through their children than they do as stand-alone entities. Look for `project_id` `86` as a notable example where the project is just a point and which has children that have far flung points in other districts. There are also projects which have no geometry themselves, but have children with geometry. These will be depicted with an array with a single null element for the `self` list and the appropriate council districts in the `self + children` list. 
* Do you like that the new column is exposed via the graphql endpoint?
* Spin up the editor, and observe that you can add and remove this column from the project list
* Do you like where I put it in the list, at the end?  How about the naming and the render() function?
* You should be able to wipe out your local storage key/value `mopedColumnConfig` and observe that it's off by default.



---
#### Ship list
